app/controllers/admin_controller.rb

```diff
@@ -16,4 +16,18 @@ def sort_column
   def sort_direction
     %w[asc desc].include?(params[:direction]) ? params[:direction] : "asc"
   end
+
+  def authorize_admin
+    if current_user.role == 3
+      redirect_to "/admin", alert: "Restricted access: please wait for an admin to approve your account"
+    end
+  end
+
+  def authorize_users
+    if current_user.role == 1 || current_user == @user
```

Comment:
Couple of alternatives to avoid the empty "if" branch:

`if current_user.role != UserRole::ADMIN && current_user != @user`

`unless current_user == UserRole::ADMIN || current_user == @user`